### PR TITLE
Move QgsVectorLayerProperties and dependencies to GUI

### PR DESCRIPTION
**Pull request by manisandro**
As was recently done with the raster layer properties dialog.

Only caevat is that the digitizing page is so tightly bound to the geometry checker code in the analysis lib that it cannot be reasonably moved to qgisgui, hence it is split off into a separate properties page using the `QgsMapLayerConfigWidget` architecture and registered in `QgisApp`. Similarly, the 3D style page was also migrated to the `QgsMapLayerConfigWidget` architecture. (Splitting out the pages probably isn't actually a bad thing per-se as it makes things more modular).

### CODE REVIEW

File: python/gui/auto_generated/vector/qgsvectorlayerproperties.sip.in [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+%Docstring
+Returns the display name entered in the dialog
+%End
+
+    void addPropertiesPageFactory( QgsMapLayerConfigWidgetFactory *factory );
+%Docstring
+Adds a properties page factory to the vector layer properties dialog.
+%End
+
+  public slots:
```
Comment (nyalldawson) [inline python/gui/auto_generated/vector/qgsvectorlayerproperties.sip.in]: There's a lot exposed here which shouldn't be -- I'd remove everything from the sip bindings except for the constructor.

**Comment by manisandro** [comment]
Any objections here if I go ahead and merge?

**Comment by nyalldawson** [comment]
I'd like to review first (but am away till mid next week)

**Comment by manisandro** [comment]
No worries, I'll hold then.

**Comment by manisandro** [comment]
@nyalldawson I'd like to merge this soonish before the code diverges too much during the 3.14 development cycle (and I suppose the earlier in the cycle it is merged the better). Any objections?

**Comment by 3nids** [comment]
Can you explain the reasoning to expose this to the API?
Why can't we rather use something in QgisInterface?

**Comment by manisandro** [comment]
I want to re-use the vector layer properties dialog from a standalone application. (This has already been done for the raster layer properties dialog, see 30569bce1f5e69bb1e2fb0e26ece78b69c8808ae)

**Comment by 3nids** [comment]
why not using QgisInterface for this?

**Comment by manisandro** [comment]
In the case of a standalone C++ application, you can only use what's exposed in qgiscore and qgisgui.

**Comment by 3nids** [comment]
My bad, thanks for the explanation

File: python/gui/auto_generated/vector/qgsvectorlayerproperties.sip.in [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+%TypeHeaderCode
+#include "qgsvectorlayerproperties.h"
+%End
+  public:
+    enum StyleType
+    {
+      QML,
+      SLD,
+      DB,
+    };
```
Comment (nyalldawson) [inline python/gui/auto_generated/vector/qgsvectorlayerproperties.sip.in]: Can we hide this from sip, and only expose the bare minimum here?

File: python/gui/auto_generated/vector/qgsvectorlayerproperties.sip.in [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+  protected slots:
+    virtual void optionsStackedWidget_CurrentChanged( int index );
+
+
+  signals:
+
+    void toggleEditing( QgsMapLayer * );
+
+    void exportAuxiliaryLayer( QgsAuxiliaryLayer *layer );
```
Comment (nyalldawson) [inline python/gui/auto_generated/vector/qgsvectorlayerproperties.sip.in]: Hide these too

**Comment by nyalldawson** [comment]
Looks good to me!

**Comment by nyalldawson** [comment]
Actually, I really like the cleanups you've done pushing the widgets to factories, nice work there...

**Comment by manisandro** [comment]
@nyalldawson Thanks for your feedback!

**Comment by 3nids** [comment]
fix in #34959

**Comment by PeterPetrik** [comment]
thanks!

**Pull request merged**